### PR TITLE
[retirement archiver] Make S3 credentials optional

### PR DESCRIPTION
**Pull request by pwnage101**
This will be necessary for systems which do not require s3 credentials
to be explicitly passed (i.e. they are inferred from the environment).

Also:

* Synchronize pylintrc with the latest version.
* Update all yaml.load() calls to yaml.safe_load() to make pylint happy.
* Update instances of len() inside if-conditions to make pylint happy.
* Remove instances of ignoring pylint redefined-variable-type.

PLAT-2418

### CODE REVIEW

File: tubular/scripts/retirement_archive_and_cleanup.py [hunk 1]
```diff
@@ -84,9 +84,15 @@ def _upload_to_s3(config, filename):
...
+        if 'access_key' in config['s3_archive']:
+            s3_credentials = (
+                config['s3_archive']['access_key'],
+                config['s3_archive']['secret_key'],
+            )
+
         s3_connection = S3Connection(
-            config['s3_archive']['access_key'],
-            config['s3_archive']['secret_key'],
+            *s3_credentials,
```
Comment (pwnage101) [inline tubular/scripts/retirement_archive_and_cleanup.py:95]: This is the meat of the change.

File: tubular/tests/test_retirement_archive_and_cleanup.py [hunk 2]
```diff
@@ -74,14 +74,16 @@ def fake_learners_to_retire():
 
 
 @patch('tubular.edx_api.BaseApiClient.get_access_token', return_value=('THIS_IS_A_JWT', None))
-@patch('tubular.scripts.retirement_archive_and_cleanup._upload_to_s3')
+@patch('tubular.scripts.retirement_archive_and_cleanup.S3Connection')
+@patch('tubular.scripts.retirement_archive_and_cleanup.Key')
```
Comment (pwnage101) [inline tubular/tests/test_retirement_archive_and_cleanup.py:78]: Updating this test was not strictly necessary, but I wanted increased coverage in the area of the code that I touched.

**Review by bmedx — APPROVED** [review]
lgtm

**Comment by doctoryes** [comment]
:+1:

**Review by doctoryes — APPROVED** [review]

**Review by macdiesel — APPROVED** [review]

**Pull request merged**